File: shared/models-js/contracts.js
// Shared JS contracts for NIRIX Diagnostics
// These mirrors shared/api-contracts/schemas and are used by the Android client.

export const AuthUser = {
  id: 'number',
  email: 'string',
  name: 'string',
};

export const Vehicle = {
  id: 'number',
  name: 'string',
  code: 'string',
};

export const TestDefinition = {
  id: 'number',
  name: 'string',
  identifier: 'string',
  test_type: 'string',
  vehicle_id: 'number',
};

export const LogEntry = {
  id: 'number',
  vehicle_id: 'number',
  test_id: 'number',
  status: 'string',
  message: 'string',
};

export const VersionInfo = {
  id: 'number',
  version: 'string',
  vehicle_id: 'number',
};

File: shared/models-py/contracts.py
"""Shared Pydantic models for NIRIX Diagnostics.

These mirror the JSON schemas under shared/api-contracts/schemas.
Backend and tools can import from here to stay in sync.
"""

from pydantic import BaseModel
from typing import Optional, List, Any


class AuthUser(BaseModel):
    id: int
    email: str
    name: str


class Vehicle(BaseModel):
    id: int
    name: str
    code: str


class TestDefinition(BaseModel):
    id: int
    name: str
    identifier: str
    test_type: str
    vehicle_id: int


class LogEntry(BaseModel):
    id: int
    vehicle_id: int
    test_id: int
    status: str
    message: str


class VersionInfo(BaseModel):
    id: int
    version: str
    vehicle_id: int
